simulation/model.py

```diff
@@ -207,13 +209,74 @@ def increment_timestep(self):
         self.households_heat_pump_aware_at_current_step = 0
 
 
+def construct_population_awareness(
+    household_population: pd.DataFrame, heat_pump_awareness: float
+) -> List[bool]:
+    """Construct a list assigning heat pump awareness to each agent in the population"""
+    # Randomly assign heat pump awareness to the agent population with a probability given by `heat_pump_awareness`
+    return [
+        random.random() < heat_pump_awareness for _ in range(len(household_population))
+    ]
+
+
+def construct_population_awareness_due_to_campaign(
+    population_heat_pump_awareness: List[bool],
+    heat_pump_awareness_due_to_campaign: float,
+    heat_pump_awareness: float,
+) -> List[bool]:
+    """Construct a list assigning heat pump awareness to each agent in the population after the campaign"""
+
+    # Awareness cannot decrease over time
```

Comment:
This comment and implementation is unclear to me.

The comment `#Awareness cannot decrease over time` suggests that these variables are 'changing' over time but isn't `heat_pump_awareness_due_to_campaign` and input to the simulation (i.e. the target) so it can't really change over time. What changes over time is the `awareness _at_time_step`?

I think the code will be a lot clearer if this logic is elsewhere (possibly right at the beginning of the script where simulation inputs are parsed). Options:
1) or you output a "warning" to in the log to say `heat_pump_awareness_due_to_campaign` value is overriden with `heat_pump_awareness` because value is lower or error or
2) don't let the simulation run and warn the users of invalid simulation input

I think the soft overriding here is very implicit and unclear and I'm trying to guess what you're doing.